### ISSUE TITLE
[RyuJIT/arm32] Assertion failed 'isGeneralRegister(id->idReg2())'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#11848]
At emitarm.cpp Line: 203

Example:
```
Assert failure(PID 12720 [0x000031b0], Thread: 9272 [0x2438]): Assertion failed 'isGeneralRegister(id->idReg2())' in 'StructABI:Main():int' (IL size 2040)

    File: c:\gh\coreclr\src\jit\emitarm.cpp Line: 203
    Image: c:\brucefo\tests\Windows_NT.arm.Checked\Tests\Core_Root\CoreRun.exe
```

Tests with this assert:
```
JIT\Directed\StructABI\StructABI\StructABI.cmd
```

Note: this test also asserts in legacy JIT:
```
Assert failure(PID 12720 [0x000031b0], Thread: 9272 [0x2438]): Assertion failed '!(op1->gtFlags & GTF_REG_VAL)' in 'StructABI:Main():int' (IL size 2040)

    File: c:\gh\coreclr\src\jit\codegenlegacy.cpp Line: 11105
    Image: c:\brucefo\tests\Windows_NT.arm.Checked\Tests\Core_Root\CoreRun.exe
```

**Comment by hqueue** [comment]
This test is skipped when running with `runtests.sh` on Linux arm32, because `JIT/Directed/StructABI/StructABI/StructABI.sh` is listed at https://github.com/dotnet/coreclr/blob/master/tests/testsFailingOutsideWindows.txt#L4

**Comment by RussKeldorph** [comment]
Dup of #8093?

**Comment by BruceForstall** [comment]
Yes.

**Comment by BruceForstall** [comment]
(Well, assuming the fix for #8093 fixes both legacy_backend and RyuJIT)
